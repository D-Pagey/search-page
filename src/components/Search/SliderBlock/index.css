.slider-block {
  background-color: #fff;
  border: 1px solid #e4e4e4;
  border-radius: 6px;
  padding: 8px 16px 4px;
}

.subscription-title {
  font-weight: bold;
  margin: 0;
}

.subscription-form {
  margin-top: 5px;
}

.rental-options {
  display: flex;
  margin-bottom: 8px;
}

.radio {
  margin-top: 4px;
}

.subscription-label {
  line-height: 1.5;
  padding-left: 8px;
}

.duration-container {
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 12px 0 10px;
}

.duration-title {
  margin: 0;
  font-size: 0.9rem;
  width: 5.25rem;
}

.months {
  background-color: white;
  border-radius: 5px;
  font-size: 1rem;
  height: 2rem;
  line-height: 1.5;
  margin: 0 8px;
  padding-left: 7px;
}
