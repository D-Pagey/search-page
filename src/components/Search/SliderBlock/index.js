import React from 'react';

import './index.css';

export default function SliderBlock() {
  return (
    <div className="subscription-container">
      <p className="subscription-title">Choose your subscription type:</p>
      <form className="subscription-form">
        <div className="rental-options">
          <div className="select-input-wrapper">
            <input type="radio" name="type-of-booking" id="1" value="on" />
            <span className="checkmark" />
          </div>
          <label className="label" htmlFor="1">
            Monthly Rolling Subscription (cancel or swap monthly)
          </label>
        </div>

        <div className="rental-options">
          <div className="select-input-wrapper">
            <input type="radio" name="type-of-booking" id="2" value="on" />
            <span className="checkmark" />
          </div>
          <label className="label" htmlFor="2">
          Minimum Commitment Subscriptions (get discounts!)
          </label>
        </div>

        <div className="subscription-duration">
          <p className="copy">Subscription Duration</p>
          <span>12 Months</span>
        </div>
      </form>
    </div>
  );
}
