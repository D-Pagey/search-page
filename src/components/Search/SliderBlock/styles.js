import styled from 'styled-components';

export const SliderBlockWrapper = styled.div`
  background-color: #fff;
  border: 1px solid #e4e4e4;
  border-radius: 6px;
  padding: 8px 16px 4px;
  @media (min-width: 768px) {
    width: 100%;
  }
`;

export const SubscriptionTitle = styled.p`
  font-weight: bold;
  margin: 0;
`;

export const SubscriptionForm = styled.form`
  margin-top: 5px;
`;

export const RentalOptionsWrapper = styled.div`
  display: flex;
  margin-bottom: 8px;
`;

export const RadioButton = styled.input.attrs({
  type: 'radio',
})`
  margin-top: 4px;
`;

export const SubscriptionLabel = styled.label`
  line-height: 1.5;
  padding-left: 8px;
`;

export const DurationWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 12px 0 10px;
`;

export const DurationTitle = styled.label`
  margin: 0;
  font-size: 0.9rem;
`;

export const MonthsDropdown = styled.select`
  background-color: white;
  border-radius: 5px;
  font-size: 1rem;
  height: 2rem;
  line-height: 1.5;
  margin: 0 8px;
  padding-left: 7px;
`;
