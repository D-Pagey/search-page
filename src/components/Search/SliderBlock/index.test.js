import React from 'react';
import { shallow } from 'enzyme';
import SliderBlock from '.';

it('should render', () => {
  const wrapper = shallow(<SliderBlock 
    fetchData={() => {}}
    handleChange={() => {}}
    userParams={{
      rental_option: 'monthly',
      number_of_months: 12,
      vehicle_type: 'Consumer',
    }}
    />);
  expect(wrapper).toMatchSnapshot();
});
