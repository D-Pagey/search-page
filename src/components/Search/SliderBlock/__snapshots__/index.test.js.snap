// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="slider-block"
>
  <p
    className="subscription-title"
  >
    Choose your subscription type:
  </p>
  <form
    className="subscription-form"
  >
    <div
      className="rental-options"
    >
      <input
        className="radio"
        defaultChecked={true}
        id="1"
        name="rental_option"
        onClick={[Function]}
        type="radio"
        value="monthly"
      />
      <label
        className="subscription-label"
        htmlFor="1"
      >
        Monthly Rolling Subscription (cancel or swap monthly)
      </label>
    </div>
    <div
      className="rental-options"
    >
      <div
        className="select-input-wrapper"
      >
        <input
          className="radio"
          defaultChecked={false}
          id="2"
          name="rental_option"
          onClick={[Function]}
          type="radio"
          value="commitment"
        />
      </div>
      <label
        className="subscription-label"
        htmlFor="2"
      >
        Minimum Commitment Subscriptions (get discounts!)
      </label>
    </div>
    <div
      className="duration-container"
    >
      <label
        className="duration-title"
        htmlFor="duration"
      >
        Subscription Duration
      </label>
      <div
        className="length"
      >
        <select
          className="months"
          defaultValue={12}
          id="duration"
          name="number_of_months"
          onChange={[Function]}
        >
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="11"
          >
            11
          </option>
          <option
            value="12"
          >
            12
          </option>
        </select>
        <span>
          months
        </span>
      </div>
    </div>
    <Button
      onClick={[Function]}
    >
      Submit
    </Button>
  </form>
</div>
`;
